lab_2_tokenize_by_bpe/main.py

```diff
@@ -58,6 +107,26 @@ def train(
     :param num_merges: required number of new tokens
     :return: dictionary in the form of <preprocessed word: number of occurrences>
     """
+    if not isinstance(word_frequencies, dict) or not isinstance(num_merges, int):
+        return None
+    merges_count = 0
+    while merges_count < num_merges:
+        pair_freq = count_tokens_pairs(word_frequencies)
+        if pair_freq is None:
+            return None
+        if not len(pair_freq) == 0:
+            sorted_dict = dict(
+                sorted(pair_freq.items(),
+                       key=lambda item: (-item[1], -len("".join(item[0])), "".join(item[0]))))
+            for k in sorted_dict.keys():
+                word_frequencies = merge_tokens(word_frequencies, k)
+                if word_frequencies is None:
+                    return None
+                merges_count += 1
+                break
```

Comment:
too many brake/continue. Revert the condition

Reply:
done